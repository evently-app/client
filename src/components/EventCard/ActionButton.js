import React from "react";
import { Animated, StyleSheet, View, TouchableOpacity, Linking } from "react-native";

import { BlurView } from "react-native-blur";

import { SCREEN_WIDTH, SCREEN_HEIGHT } from "../../lib/constants";
import { colors } from "../../lib/styles";

import { Header, SubHeader, Paragraph } from "../universal/Text";

const INITIAL_WIDTH = 100;

const ActionButton = ({ title, url, yOffset }) => {
	console.log("URL IS: ", url)
	const inputRange = [0, 50, 110, 150];

	const animatedStyle = {
		overflow: "hidden",
		borderRadius: yOffset.interpolate({
			inputRange,
			outputRange: [10, 10, 10, 0],
			extrapolate: "clamp"
		}),
		transform: [
			{
				translateX: yOffset.interpolate({
					inputRange,
					outputRange: [0, -25, -55, -110],
					extrapolate: "clamp"
				})
			},
			{
				translateY: yOffset.interpolate({
					inputRange,
					outputRange: [0, -1.25, -2.5, -5],
					extrapolate: "clamp"
				})
			},
			{
				scaleX: yOffset.interpolate({
					inputRange,
					outputRange: [
						1,
						SCREEN_WIDTH / (4 * INITIAL_WIDTH),
						SCREEN_WIDTH / (2 * INITIAL_WIDTH),
						SCREEN_WIDTH / INITIAL_WIDTH
					],
					extrapolate: "clamp"
				})
			},
			{
				scaleY: yOffset.interpolate({
					inputRange,
					outputRange: [1, 1, 1.5, 2],
					extrapolate: "clamp"
				})
			}
		]
	};

	const animatedTextStyle = {
		transform: [
			{
				translateX: yOffset.interpolate({
					inputRange,
					outputRange: [0, -23.5, -50, -100],
					extrapolate: "clamp"
				})
			},
			{
				translateY: yOffset.interpolate({
					inputRange,
					outputRange: [0, -1.25, -2.5, -5],
					extrapolate: "clamp"
				})
			}
		]
	};

	return (
		<TouchableOpacity activeOpacity={0.9} 
			onPress={() => {Linking.openURL(url);}}>
			<Animated.View style={[styles.container, animatedStyle]}>
				<BlurView blurType="xlight" style={styles.fill} />
			</Animated.View>

			<View style={styles.text}>
				<SubHeader animated style={{ ...animatedTextStyle, color: "black" }}>
					{title}
				</SubHeader>
			</View>
		</TouchableOpacity>
	);
};

const styles = StyleSheet.create({
	container: {
		right: 10,
		bottom: 10,
		width: 150,
		height: 30,
		position: "absolute"
		// borderRadius: 10
	},
	fill: {
		position: "absolute",
		top: 0,
		bottom: 0,
		left: 0,
		right: 0
	},
	text: {
		position: "absolute",
		right: 25,
		bottom: 10,
		height: 30,
		// paddingHorizontal: 10,
		// width: 300,
		justifyContent: "center",
		alignItems: "center"
	}
});

export default ActionButton;
