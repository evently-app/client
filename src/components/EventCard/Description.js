import React from "react";
import { Animated, StyleSheet, View, Text} from "react-native";

import { SCREEN_WIDTH, SB_HEIGHT, SCREEN_HEIGHT } from "../../lib/constants";
import { colors, fonts } from "../../lib/styles";

const description = ({ description }) => {
	// Add processing here for if description is too long

	return (
		<View style={styles.container}>
			<Text style={styles.description}>
				{description}
			</Text>
		</View>
	);
};

const styles = StyleSheet.create({
	container: {
		backgroundColor: colors.lightpurple,
		width: SCREEN_WIDTH - 20,
		height: SCREEN_HEIGHT - 200,
		padding: 20,
	},
	description: {
		color: "white",
		textAlign: 'left',
		fontFamily: fonts.descriptions,
	},
});

export default description;
