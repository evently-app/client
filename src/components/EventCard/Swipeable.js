import React, { Component } from "react";
import { StyleSheet, Animated } from "react-native";

import Interactable from "react-native-interactable";
import Haptics from "react-native-haptic-feedback";

import { SCREEN_HEIGHT, SCREEN_WIDTH } from "../../lib/constants";

class Swipeable extends Component {
	deltaX = new Animated.Value(0);

	handleOnSnap = ({ nativeEvent }) => {
		Haptics.trigger("impactLight");
		const { index } = nativeEvent;
		const { onSwipeRight, onSwipeLeft } = this.props;

		if (index === 0) {
			onSwipeLeft();
		} else if (index === 1) {
		} else if (index === 2) {
			onSwipeRight();
		}
	};

	render() {
		const left = { x: -400 };
		const centered = { x: 0 };
		const right = { x: 400 };

		const animatedRotation = {
			transform: [
				{
					rotate: this.deltaX.interpolate({
						inputRange: [-300, 0, 300],
						outputRange: ["-10deg", "0deg", "10deg"]
					})
				}
			]
		};

		return (
			<Interactable.View
				animatedNativeDriver
				style={[animatedRotation, styles.container]}
				ref={view => (this.interactable = view)}
				horizontalOnly={true}
				snapPoints={[left, centered, right]}
				onSnap={this.handleOnSnap}
				onDrag={this.handleOnDrag}
				initialPosition={centered}
				animatedValueX={this.deltaX}
			>
				{this.props.children}
			</Interactable.View>
		);
	}
}

const styles = StyleSheet.create({
	container: {
		position: "absolute",
		justifyContent: "center",
		alignItems: "center"
	}
});

export default Swipeable;
