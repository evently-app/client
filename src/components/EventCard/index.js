import React, { Component } from "react";
import { View, Text } from "react-native";

import { SCREEN_WIDTH, SCREEN_HEIGHT } from "../../lib/constants";

class EventCard extends Component {
	render() {
		return (
			<View
				style={{
					width: SCREEN_WIDTH - 20,
					height: SCREEN_HEIGHT - 100,
					backgroundColor: this.props.backgroundColor,
					borderRadius: 20
				}}
			>
				<Text>{this.props.title}</Text>
			</View>
		);
	}
}

export default EventCard;
