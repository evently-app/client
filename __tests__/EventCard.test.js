import "react-native";
import React from "react";
import EventCard from "../src/components/EventCard";
import renderer from "react-test-renderer";
import ActionButton from "../src/components/EventCard/ActionButton";
import Description from "../src/components/EventCard/Description";

jest.mock("@mapbox/react-native-mapbox-gl");

const yOffset = {
	interpolate: () => {}
};

test("EventCard renders correctly", () => {
	const tree = renderer.create(<EventCard />).toJSON();
	expect(tree).toMatchSnapshot();
});

it("Description renders correctly", () => {
	const snap = shallow(<Description />);
	expect(snap).toMatchSnapshot();
});

it("Action Button renders correctly", () => {
	const snap = shallow(<ActionButton yOffset={yOffset} />);
	expect(snap).toMatchSnapshot();
});
