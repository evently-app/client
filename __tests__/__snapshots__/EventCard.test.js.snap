// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Button renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.9}
  onPressIn={[Function]}
>
  <AnimatedComponent
    style={
      Array [
        Object {
          "bottom": 10,
          "height": 30,
          "position": "absolute",
          "right": 10,
          "width": 112,
        },
        Object {
          "borderRadius": undefined,
          "overflow": "hidden",
          "transform": Array [
            Object {
              "translateX": undefined,
            },
            Object {
              "translateY": undefined,
            },
            Object {
              "scaleX": undefined,
            },
            Object {
              "scaleY": undefined,
            },
          ],
        },
      ]
    }
  >
    <BlurView
      blurAmount={10}
      blurType="xlight"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </AnimatedComponent>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 10,
        "height": 30,
        "justifyContent": "center",
        "position": "absolute",
        "right": 25,
      }
    }
  >
    <SubHeader
      animated={true}
      style={
        Object {
          "color": "black",
          "transform": Array [
            Object {
              "translateX": undefined,
            },
            Object {
              "translateY": undefined,
            },
          ],
        }
      }
    />
  </View>
</TouchableOpacity>
`;

exports[`Description renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(52, 33, 79, 0.95)",
      "height": 1134,
      "padding": 20,
      "width": 730,
    }
  }
>
  <Paragraph />
</View>
`;

exports[`EventCard renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(21,2,24,0.95)",
      "borderRadius": 20,
      "height": 1154,
      "overflow": "hidden",
      "width": 730,
    }
  }
>
  <RCTScrollView
    bounces={false}
    onScroll={[Function]}
    scrollEventThrottle={16}
    showsVerticalScrollIndicator={false}
  >
    <View>
      <Image
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "alignItems": "center",
            "height": 1134,
            "justifyContent": "center",
            "width": 730,
          }
        }
      />
      <BVLinearGradient
        colors={
          Array [
            3003121664,
            436207616,
          ]
        }
        endPoint={
          Object {
            "x": 0.5,
            "y": 1,
          }
        }
        locations={
          Array [
            0,
            0.1,
          ]
        }
        startPoint={
          Object {
            "x": 0.5,
            "y": 0,
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <View
        style={
          Object {
            "padding": 10,
            "position": "absolute",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Avenir Next",
              "fontSize": 20,
              "fontWeight": "700",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Avenir Next",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Sun April 14th 7:44 pm
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(52, 33, 79, 0.95)",
            "height": 1134,
            "padding": 20,
            "width": 730,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Avenir Next",
              "fontSize": 12,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 10,
          "bottom": 10,
          "height": 30,
          "overflow": "hidden",
          "position": "absolute",
          "right": 10,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
            Object {
              "scaleX": 1,
            },
            Object {
              "scaleY": 1,
            },
          ],
          "width": 112,
        }
      }
    >
      <BlurView
        blurAmount={10}
        blurType="xlight"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "bottom": 10,
          "height": 30,
          "justifyContent": "center",
          "position": "absolute",
          "right": 25,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Avenir Next",
            "fontSize": 16,
            "fontWeight": "600",
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        Get Tickets
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "gray",
        "borderRadius": 2.5,
        "height": 100,
        "position": "absolute",
        "right": 10,
        "top": 10,
        "width": 5,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "lightgray",
          "borderRadius": 2.5,
          "height": 20,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
          "width": 5,
        }
      }
    />
  </View>
</View>
`;
